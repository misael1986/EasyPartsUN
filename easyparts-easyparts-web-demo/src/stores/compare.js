import { decorate, observable } from "mobx";

class Compare {
  constructor() {
    this.parts = [];
  }
}

decorate(Compare, {
  parts: observable,
});

export default new Compare();
