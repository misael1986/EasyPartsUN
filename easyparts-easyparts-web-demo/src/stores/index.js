export { default as SearchStore } from "./search";
export { default as UserStore } from "./user";
export { default as CompareStore } from "./compare";
export { default as PurposeStore } from "./purpose";
