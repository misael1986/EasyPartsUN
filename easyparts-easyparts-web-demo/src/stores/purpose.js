import { decorate, observable } from "mobx";

class Purpose {
  constructor() {
    this.name = null;
    this.list = [];
  }
}

decorate(Purpose, {
  name: observable,
  list: observable,
});

export default new Purpose();
