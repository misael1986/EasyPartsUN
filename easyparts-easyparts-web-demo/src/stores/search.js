import { decorate, observable } from "mobx";

class Search {
  constructor() {
    this.cpuList = [];
    this.motherboardList = [];
    this.gpuList = [];
    this.ramList = [];
    this.storage1List = [];
    this.storage2List = [];
    this.chassisList = [];
    this.powerSourceList = [];

    this.typeSelected = "cpu";

    this.searchList = [];
    this.searchText = "";
    this.componentDescription = "Procesador";

    this.cpuFilters = [];
    this.motherboardFilters = [];
    this.gpuFilters = [];
    this.ramFilters = [];
    this.storage1Filters = [];
    this.storage2Filters = [];
    this.chassisFilters = [];
    this.powerSourceFilters = [];
  }

  clear() {
    this.cpuList = [];
    this.motherboardList = [];
    this.gpuList = [];
    this.ramList = [];
    this.storage1List = [];
    this.storage2List = [];
    this.chassisList = [];
    this.powerSourceList = [];

    this.typeSelected = "cpu";

    this.searchList = [];
    this.searchText = "";
    this.componentDescription = "Procesador";

    this.cpuFilters = [];
    this.motherboardFilters = [];
    this.gpuFilters = [];
    this.ramFilters = [];
    this.storage1Filters = [];
    this.storage2Filters = [];
    this.chassisFilters = [];
    this.powerSourceFilters = [];
  }

  get filterFromType() {
    return this[this.typeSelected + "Filters"];
  }

  get searchListFiltered() {
    let byName = this.searchList.filter((it) => {
      return it.nombre.includes(this.searchText);
    });
    const filtersForType = this.filterFromType;
    for (let i = 0; i < filtersForType.length; i++) {
      byName = byName.filter((it) => {
        return !(
          filtersForType[i].selected !== null &&
          it[filtersForType[i].databaseName] !== filtersForType[i].selected
        );
      });
    }
    return byName;
  }
}

decorate(Search, {
  searchList: observable,
  searchText: observable,

  componentDescription: observable,

  cpuFilters: observable,
  motherboardFilters: observable,
  gpuFilters: observable,
  ramFilters: observable,
  storage1Filters: observable,
  storage2Filters: observable,
  chassisFilters: observable,
  powerSourceFilters: observable,
});

export default new Search();
