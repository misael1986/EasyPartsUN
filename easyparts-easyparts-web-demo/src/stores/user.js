import { computed, decorate, observable } from "mobx";

class User {
  constructor() {
    this.email = "";
    this.password = "";
    this.otp = "";
    this.loggedIn = false;

    this.pc = {
      chassis: null,
      cpu: null,
      gpu: null,
      motherboard: null,
      powerSource: null,
      ram: null,
      storage1: null,
      storage2: null,
    };

    this.quantities = {
      ram: 0,
      storage1: 0,
      storage2: 0,
      gpu: 0,
    };

    this.pc2 = {
      chassis: null,
      cpu: null,
      gpu: null,
      motherboard: null,
      powerSource: null,
      ram: null,
      storage1: null,
      storage2: null,
    };

    this.quantities2 = null;

    this.routeOrigin = "/";
  }

  setRamQuantity(increase) {
    if (increase && this.quantities.ram === 4) return;
    if (!increase && this.quantities.ram === 1) return;

    this.quantities = {
      ...this.quantities,
      ram: (this.quantities.ram = increase
        ? this.quantities.ram * 2
        : this.quantities.ram / 2),
    };
  }

  setStorageQuantity(id, increase) {
    const prev = this.quantities["storage" + id];
    if (increase && prev > 2) return;
    if (!increase && prev === 1) return;

    if (increase && this.pc["storage" + id].tecnologia === "M2") return;

    this.quantities = {
      ...this.quantities,
      ["storage" + id]: (this.quantities["storage" + id] += increase ? 1 : -1),
    };
  }

  get pcPower() {
    let total = 0;
    Object.keys(this.pc).forEach((it) => {
      if (this.pc[it] !== null && this.pc[it].potencia_max !== undefined) {
        total += this.pc[it].potencia_max;
      }
    });
    return total;
  }

  getPcPrice(pc, quantities) {
    let total = 0;
    Object.keys(pc).forEach((it) => {
      if (pc[it] !== null && pc[it].precio !== undefined) {
        if (quantities[pc[it].type] !== undefined) {
          total += pc[it].precio * quantities[pc[it].type];
        } else {
          total += pc[it].precio;
        }
      }
      //console.log(toJS(pc[it]));
      //console.log("New total with type: " + pc[it].type + " = " + total);
    });
    return total;
  }

  clear() {
    this.email = "";
    this.password = "";
    this.otp = "";
    this.loggedIn = false;

    this.clearPc();
    this.clearPc2();

    this.routeOrigin = "/";
  }

  clearPc() {
    this.pc = {
      chassis: null,
      cpu: null,
      gpu: null,
      motherboard: null,
      powerSource: null,
      ram: null,
      storage1: null,
      storage2: null,
    };

    this.quantities = {
      ram: 0,
      storage1: 0,
      storage2: 0,
      gpu: 0,
    };
  }

  clearPc2() {
    this.pc2 = {
      chassis: null,
      cpu: null,
      gpu: null,
      motherboard: null,
      powerSource: null,
      ram: null,
      storage1: null,
      storage2: null,
    };

    this.quantities2 = {
      ram: 0,
      storage1: 0,
      storage2: 0,
      gpu: 0,
    };
  }
}

decorate(User, {
  email: observable,
  password: observable,
  otp: observable,
  loggedIn: observable,

  pc: observable,
  pcPower: computed,

  quantities: observable,
});

export default new User();
