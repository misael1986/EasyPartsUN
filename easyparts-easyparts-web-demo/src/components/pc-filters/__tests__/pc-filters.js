import React from "react";
import renderer from "react-test-renderer";
import PcFilters from "../pc-filters";
import { Provider } from "mobx-react";
import { SearchStore } from "../../../stores";

describe("PcFilters", () => {
  it("hasn't changed unexpectedly", () => {
    const tree = renderer
      .create(
        <Provider searchStore={SearchStore}>
          <PcFilters
            filters={[{ values: [1, 2] }]}
            selectFilter={() => {}}
            resetFilters={() => {}}
          />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
