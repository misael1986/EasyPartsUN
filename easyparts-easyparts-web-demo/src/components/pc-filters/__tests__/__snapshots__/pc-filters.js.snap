// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PcFilters hasn't changed unexpectedly 1`] = `
<div
  className="PcFilters"
>
  <div>
    <button
      className="top"
      onClick={[Function]}
    >
      <p
        className="topText"
      />
      <svg
        color="white"
        fill="currentColor"
        height={14}
        size={14}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "white",
          }
        }
        viewBox="0 0 512 512"
        width={14}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zm116-292H256v-70.9c0-10.7-13-16.1-20.5-8.5L121.2 247.5c-4.7 4.7-4.7 12.2 0 16.9l114.3 114.9c7.6 7.6 20.5 2.2 20.5-8.5V300h116c6.6 0 12-5.4 12-12v-64c0-6.6-5.4-12-12-12z"
        />
      </svg>
    </button>
    <div
      className="List"
    >
      <button
        className="listItem"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="listItem"
        onClick={[Function]}
      >
        2
      </button>
    </div>
  </div>
  <button
    className="reset"
    onClick={[Function]}
  >
    <svg
      baseProfile="tiny"
      color="white"
      fill="currentColor"
      height={26}
      size={26}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "white",
        }
      }
      version="1.2"
      viewBox="0 0 24 24"
      width={26}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12.872 13.191h5.128v-5.127c-.008-1.135-.671-1.408-1.473-.605l-1.154 1.158c-1.015-.795-2.257-1.23-3.566-1.23-1.55 0-3.009.604-4.104 1.701-1.099 1.092-1.703 2.553-1.703 4.103 0 1.553.604 3.012 1.701 4.107 1.097 1.097 2.555 1.702 4.106 1.702 1.55 0 3.009-.605 4.106-1.703.296-.297.558-.621.78-.965.347-.541.19-1.26-.35-1.605-.539-.346-1.258-.189-1.604.35-.133.207-.292.4-.468.58-.659.658-1.534 1.02-2.464 1.02-.93 0-1.805-.361-2.464-1.02-.657-.658-1.02-1.533-1.02-2.465 0-.93.362-1.805 1.02-2.461.659-.658 1.534-1.021 2.464-1.021.688 0 1.346.201 1.909.572l-1.448 1.451c-.803.802-.53 1.458.604 1.458z"
      />
    </svg>
    <p
      className="resetText"
    >
      Resetear
    </p>
  </button>
</div>
`;
